static/app/components/modals/metricWidgetViewerModal.tsx

```diff
@@ -227,13 +228,16 @@ function MetricWidgetViewerModal({
   return (
     <Fragment>
       <OrganizationContext.Provider value={organization}>
-        <Header closeButton>
+        <Header>
           <MetricWidgetTitle
             title={title}
             onTitleChange={handleTitleChange}
             placeholder={widgetMQL}
             description={widget.description}
           />
+          <div onClick={handleSubmit}>
+            <CloseButton />
+          </div>
```

Comment:
Why not put it on the close button?

Reply:
asking because an `onClick` on a button is also triggered for keyboard interaction.

Reply:
Because close button overrides whatever onClick handler is passed to it :/ I added a comment. We should move this to a lideout anyways soon